include/dt-bindings/gpio/gpio.h

```diff
@@ -1,180 +1,429 @@
+ */
+#define GPIO_INT_LOW    (1 << (GPIO_INT_POS + 2))
+
+/** Trigger detection on input state is (or transitions to) high.
+ *
+ * This is a component flag that should be combined with other
+ * `GPIO_INT_*` flags to produce a meaningful configuration.
+ */
+#define GPIO_INT_HIGH   (1 << (GPIO_INT_POS + 3))
+
+/** Interrupt triggers on both rising and falling edge.
+ *
+ * This is a component flag that should be combined with other
+ * `GPIO_INT_*` flags to produce a meaningful configuration.
+ *
+ * For compatibility with old code this should be combined with
+ * `GPIO_INT_EDGE`.
+ *
+ * For compatibility with new code this should be combined with both
+ * `GPIO_INT_LOW` and `GPIO_INT_HIGH`.  Updated driver code should
+ * prefer to check the individual flags.
+ *
+ * @deprecated Replace tests with `GPIO_INT_LOW | GPIO_INT_HIGH`.
+ *
+ * @{
+ */
+#define GPIO_INT_DOUBLE_EDGE    (1 << (GPIO_INT_POS + 4))
+
+/** GPIO pin trigger on rising edge. */
+#define GPIO_INT_EDGE_RISING    (GPIO_INT_ENABLE \
+				 | GPIO_INT_EDGE \
+				 | GPIO_INT_HIGH)
+
+/** GPIO pin trigger on rising edge. */
+#define GPIO_INT_EDGE_FALLING   (GPIO_INT_ENABLE \
+				 | GPIO_INT_EDGE \
+				 | GPIO_INT_LOW)
+
+/** GPIO pin trigger on rising or falling edge. */
+#define GPIO_INT_EDGE_BOTH      (GPIO_INT_ENABLE \
+				 | GPIO_INT_EDGE \
+				 | GPIO_INT_LOW	 \
+				 | GPIO_INT_HIGH \
+				 | GPIO_INT_DOUBLE_EDGE)
+
+/** GPIO pin trigger on level low. */
+#define GPIO_INT_LEVEL_LOW      (GPIO_INT_ENABLE  \
+				 | GPIO_INT_LEVEL \
+				 | GPIO_INT_LOW)
+
+/** GPIO pin trigger on level high. */
+#define GPIO_INT_LEVEL_HIGH     (GPIO_INT_ENABLE  \
+				 | GPIO_INT_LEVEL \
+				 | GPIO_INT_HIGH)
+
+/** @} */
 
 /**
- * @name GPIO pull flags
- * The `GPIO_PUD_*` flags are used with `gpio_pin_configure` or `gpio_port_configure`,
- * to specify the pull-up or pull-down electrical configuration of a GPIO pin.
+ * @name GPIO polarity flags
+ *
+ * The `GPIO_POL_*` flags are used with `gpio_pin_configure` or
+ * `gpio_port_configure`, to specify the polarity of a GPIO pin.
+ *
  * @{
  */
 /** @cond INTERNAL_HIDDEN */
-#define GPIO_PUD_POS		8
+#define GPIO_POL_POS            14
+#define GPIO_POL_MASK           (1 << GPIO_POL_POS)
 /** @endcond */
 
-/** Pin is neither pull-up nor pull-down. */
-#define GPIO_PUD_NORMAL		(0 << GPIO_PUD_POS)
+/** GPIO pin polarity is normal. */
+#define GPIO_POL_NORMAL         (0 << GPIO_POL_POS)
 
-/** Enable GPIO pin pull-up. */
-#define GPIO_PUD_PULL_UP	(1 << GPIO_PUD_POS)
+/** GPIO pin polarity is inverted. */
+#define GPIO_POL_INV            (1 << GPIO_POL_POS)
 
-/** Enable GPIO pin pull-down. */
-#define GPIO_PUD_PULL_DOWN	(2 << GPIO_PUD_POS)
+/** @} */
 
 /** @cond INTERNAL_HIDDEN */
-#define GPIO_PUD_MASK		(3 << GPIO_PUD_POS)
+#define GPIO_DEBOUNCE_POS       15
+#define GPIO_DEBOUNCE_MASK      (1 << GPIO_DEBOUNCE_POS)
 /** @endcond */
-/** @} */
 
+/** Enable GPIO pin debounce. */
+#define GPIO_DEBOUNCE           (1U << GPIO_DEBOUNCE_POS)
```

Comment:
Just a detail. I think it would be good to use unsigned values in all the shifts here. To avoid problems with undefined behavior in case the bit positions are altered in the future, or some definitions are modified after copying to some other module etc.